substratevm/src/org.graalvm.polyglot.nativeapi/src/org/graalvm/polyglot/nativeapi/PolyglotNativeAPI.java

```diff
@@ -243,6 +243,21 @@ public static PolyglotStatus poly_engine_get_languages(PolyglotIsolateThread thr
             }
         });
     }
+    
+     @CEntryPoint(name = "poly_engine_get_languages_size", documentation = { 
```

Comment:
Adding this function would be redundant. The size of the array is already returned by the `SizeTPointer size`. I think only the documentation of that function needs to be updated.

Reply:
I agree it's redundant. I just saw the mention in the docs and thought it was an error.
However, this pattern for using it is also somewhat redundant and confused me at first:

```c

size_t num_languages;
poly_language languages_ptr[num_languages];

// get the number of languages
poly_engine_get_languages(thread, engine, NULL, &num_languages);

// put the actual language pointers into the array
poly_engine_get_languages(thread, engine, &languages_ptr, &num_languages); 
```

Maybe the best thing to do would be to explicitly state in the doc comment that to use it, it requires being called twice, with different combinations of parameters?

Reply:
Yes, improving the docs is best. I know it is strange, I had the same argument with the C guy when we were designing this. Apparently, this is the standard thing in C.

Reply:
> I had the same argument with the C guy when we were designing this. Apparently, this is the standard thing in C.

🤣 Glad I was not the only one confused, haha.
Awesome -- I will remove the extra method and just update the existing docs accordingly 👍